continuous-integration/readme.md

```diff
@@ -1,3 +1,236 @@
+  - We recommend incorporating security analysis tools within the build stage of your pipeline such as: code credential scanner, security risk detection, static analysis, etc. For Azure DevOPS, you can add a security scan task to your pipeline by installing the [Microsoft Security Code Analysis Extension](https://secdevtools.azurewebsites.net/#pills-onboard). Github Actions supports a similar extension with the [RIPS security scan solution](https://github.com/marketplace/actions/rips-security-scan).
+  - Code standards are maintained within a single configuration file. There should be a step in your build pipeline that asserts code in the latest commit conforms to the known style definition.
+- [ ] **Build Script Target**
+  - A single command should have the capability of building the system. This is also true for builds running on a CI server or on a developers local machine.
+- [ ] **No IDE Dependencies**
+  - It's essential to have a master build that's runnable through standalone scripts and not dependent on a particular IDE. Build pipeline targets can be triggered locally on their desktops through their IDE of choice. The build process should maintain enough flexibility to run within a CI server as well. As an example, dockerizing your build process offers this level of flexibility as VSCode and IntelliJ supports [docker plugin](https://code.visualstudio.com/docs/containers/overview) extensions.
+
+## Build Environment Dependencies
+
+- [ ] **Automated local environment setup**
+  - We encourage maintaining a consistent developer experience for all team members. There should be a central automated manifest / process that streamlines the installation and setup of any software dependencies. This way developers can replicate the same build environment locally as the one running on a CI server.
+  - Build automation scripts often require specific software packages and version pre-installed within the runtime environment of the OS. This presents some challenges as build processes typically version lock these dependencies.
+  - All developers on the team should be able to emulate the build environment from their local desktop regardless of their OS.
+  - Well established software packaging tools like Docker, Maven, npm, etc should be considered when designing your build automation tool chain.
+- [ ] **Document local setup**
+  - The setup process for setting up a local build environment should be well documented and easy for developers to follow.
+  
+## Infrastructure as Code
+
+Manage as much of the following as possible, as code:
+
+- Configuration Files
+- Configuration Management
```

Comment:
This is a little confusing, how do you manage configuration management and secret management as code?

Reply:
Through Terraform's keyvault provider. I'll include a reference example to help make this more clear.